### ISSUE TITLE
Set --input-encoding internally if BOM is found (Was: Handle the UTF-8 BOM)

### ISSUE THREAD

**Issue by kachelaqa** [universal-ctags/ctags#1422]
The current Unicode Standard permits the use of the UTF-8 BOM (without recoomending or requiring it), and it is quite widely used in practice (especially on Windows). Any decent modern source code editor can handle it transparently, and I think ctags should do likewise.

If ctags doesn't skip the UTF-8 BOM when reading a source code file, it may fail to find all the valid tags. For example, the IniConf parser will fail to find the first section if it is placed at the beginning of the file (because the first line will not start with a "[").

**Comment by masatake** [comment]
Could you put somewhere an example input including BOM?
Should I skip the first 3 bytes (0xEF 0xBB 0xBF) ?
It is not enough?

**Comment by k-takata** [comment]
I think skipping the first 3 bytes is enough for the first step, but maybe it's better to set the input encoding to "UTF-8" like `--input-encoding=UTF-8`. Note that the setting should be applied only to the file with BOM.

**Comment by masatake** [comment]
@k-takata, oh, I see. thank you. I think I can implement it.

**Comment by kachelaqa** [comment]
@masatake 
In older versions of universal-ctags, implementing this was very simple. It was just a matter of skipping the initial 0xEF 0xBB 0xBF in openInputFile/read.c. But in more recent versions this has become much more complicated because of all the MIO stuff. I did manage to get a very crude patch working, but I just don't know the code-base well enough to do it properly (hence this feature request). Anyway, what I found, was that it was necessary to re-skip the BOM after every use of mio.rewind(). But I'm not sure whether that alone is sufficient.
